tests/Microsoft.Identity.Web.Test/CacheExtensionsTests.cs

```diff
@@ -33,19 +37,39 @@ public void InMemoryCacheExtensionsTests()
         public async Task CacheExtensions_CcaAlreadyExists_TestsAsync()
         {
             AuthenticationResult result;
+            TestTelemetryClient testTelemetryClient = new TestTelemetryClient(TestConstants.ClientId);
             // new InMemory serializer and new cca
-            result = await CreateAppAndGetTokenAsync(CacheType.InMemory, addInstanceMock: true).ConfigureAwait(false);
+            result = await CreateAppAndGetTokenAsync(CacheType.InMemory, testTelemetryClient, addInstanceMock: true).ConfigureAwait(false);
             Assert.Equal(TokenSource.IdentityProvider, result.AuthenticationResultMetadata.TokenSource);
+            AssertCacheTelemetry(testTelemetryClient, CacheLevel.None);
 
-            result = await CreateAppAndGetTokenAsync(CacheType.InMemory, addTokenMock: false).ConfigureAwait(false);
+            result = await CreateAppAndGetTokenAsync(CacheType.InMemory, testTelemetryClient, addTokenMock: false).ConfigureAwait(false);
             Assert.Equal(TokenSource.Cache, result.AuthenticationResultMetadata.TokenSource);
+            AssertCacheTelemetry(testTelemetryClient, CacheLevel.L1Cache);
 
             // new DistributedInMemory and same cca
-            result = await CreateAppAndGetTokenAsync(CacheType.DistributedInMemory).ConfigureAwait(false);
+            result = await CreateAppAndGetTokenAsync(CacheType.DistributedInMemory, testTelemetryClient).ConfigureAwait(false);
             Assert.Equal(TokenSource.IdentityProvider, result.AuthenticationResultMetadata.TokenSource);
+            AssertCacheTelemetry(testTelemetryClient, CacheLevel.None);
 
-            result = await CreateAppAndGetTokenAsync(CacheType.DistributedInMemory, addTokenMock: false).ConfigureAwait(false);
+            result = await CreateAppAndGetTokenAsync(CacheType.DistributedInMemory, testTelemetryClient, addTokenMock: false).ConfigureAwait(false);
             Assert.Equal(TokenSource.Cache, result.AuthenticationResultMetadata.TokenSource);
+            AssertCacheTelemetry(testTelemetryClient, CacheLevel.L1Cache);
+        }
+
+        [Fact]
+        public async Task CacheExtensions_CcaAlreadyExistsL2_TestsAsync()
+        {
+            AuthenticationResult result;
+            TestTelemetryClient testTelemetryClient = new TestTelemetryClient(TestConstants.ClientId);
+            // new DistributedInMemory serializer with L1 cache disabled
+            result = await CreateAppAndGetTokenAsync(CacheType.DistributedInMemory, testTelemetryClient, addInstanceMock: true, disableL1Cache: true).ConfigureAwait(false);
```

Comment:
Consider adding variations for:

- `disableL1Cache` is `false. 
- MsalMemoryCacheProvider
- no cache (i.e. use MSAL's internal cache)